cluster-autoscaler/cloudprovider/oci/oke/oci/cache.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2020-2023 Oracle and/or its affiliates.
+*/
+
+package oci
```

Comment:
I see the `oci` package name is duplicated. For maintainability, we should organize the implementation specific code in separate packages (`oke` & `instancepools`). 

There is also common code between the two implementations that could live an a common package (`common`) so that we could re-use certain functions, constants, etc. rather than having duplicate code (`oci_util.go` is a good start).

Reply:
Yes, I agree that there is a lot of common code between the two implementations. There is a plan afterwards to integrate the two implementations more tightly. This is an initial pass to open source our internal code that we know is working.

Reply:
completed with current refactor. See latest comment. I have split things into three compartments.

1. instance pool
2. node pool
3. common